subsys/bluetooth/shell/bt.c

```diff
@@ -233,6 +299,9 @@ static void scan_recv(const struct bt_le_scan_recv_info *info,
 		    phy2str(info->primary_phy), phy2str(info->secondary_phy),
 		    info->interval, BT_CONN_INTERVAL_TO_US(info->interval),
 		    info->sid);
+	if (scan_verbose_output && strlen(scan_data.formatted_data_sets)) {
+		shell_info(ctx_shell, "          [SCAN DATA START - %s]\n%s          [SCAN DATA END]", scan_response_type_txt[info->adv_type], scan_data.formatted_data_sets);
+	}
```

Comment:
Why cannot call `bt_data_parse(buf, my_print_all_cb, NULL);` that will just print whatever you need directly to the console